debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/OffsetContextMessageProcessingIT.java

```diff
@@ -0,0 +1,575 @@
+                "  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP" +
+                ");";
+        TestHelper.execute(ddlStatement);
+        LOGGER.debug("Created test table: {}", tableName);
+    }
+
+    @After
+    public void after() throws Exception {
+        // Always stop connector and clean up slots after each test
+        try {
+            stopConnector();
+        }
+        catch (Exception e) {
+            // Ignore if connector is already stopped
+        }
+
+        // Clean up replication slots after each test
+        cleanupReplicationSlots();
+
+        // Clear any remaining consumer records to prevent test interference
+        if (consumer != null) {
+            try {
+                consumer.clear();
+            }
+            catch (Exception e) {
+                // Ignore cleanup errors
+            }
+        }
+
+        // Small delay to ensure cleanup is complete before next test
+        try {
+            Thread.sleep(100);
+        }
+        catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+    }
+
+    /**
+     * Override to use the correct server name for streaming metrics
+     */
+    @Override
+    protected void waitForStreamingToStart() throws InterruptedException {
+        // Get the current test method name to determine which server name to use
+        String testMethodName = getCurrentTestMethodName();
+        String serverName = getServerNameForTest(testMethodName);
+        waitForStreamingRunning("postgres", serverName);
+    }
+
+    private String getCurrentTestMethodName() {
+        return Thread.currentThread().getStackTrace()[3].getMethodName();
+    }
+
+    private String getServerNameForTest(String testMethodName) {
+        switch (testMethodName) {
+            case "shouldProcessAllMessagesWhenOffsetContextIsNull":
+                return "test_server_scenario1";
+            case "shouldStartFromNextInsertWhenLastProcessedWasCommit":
+                return "test_server_scenario2";
+            case "shouldResumeFromNextUnprocessedMessageInSameBatch":
+                return "test_server_scenario3";
+            case "shouldProcessAllMessagesFromIntermediateOffsetWithAdditionalStatements":
+                return "test_server_scenario4";
+            case "shouldHandleLsnComparisonAcrossMultipleRestartCycles":
+                return "test_server_scenario5";
+            // Simple message skip optimization test case
+            case "testMessageSkipOptimizationOnRestart":
+                return "test_server_skip_optimization";
+            default:
+                return "test_server"; // fallback
+        }
+    }
+
+    /**
+     * Scenario 1: Offset context is null - No message should be skipped
+     *
+     * When connector starts fresh (no previous offset), all messages should be processed
+     * from the beginning of the replication stream.
+     */
+    @Test
+    @FixFor("OFFSET-CONTEXT-SCENARIOS")
```

Comment:
please create a jira issue for this and place Jira issue number (DBZ-XYZ) here and in other places